### PR TITLE
Reduce error handling verbosity in CI tests scripts

### PR DESCRIPTION
**Pull request by AjayThorve**
This PR adds a less verbose [trap method](https://github.com/rapidsai/cugraph/blob/f2b081075704aabc789603e14ce552eac3fbe692/ci/test.sh#L19), for error handling to help ensure that we capture all potential error codes in our test scripts, and works as follows:

- setting an environment variable, EXITCODE, with a default value of 0
- setting a trap statement triggered by ERR signals which will set EXITCODE=1 when any commands return a non-zero exit code

cc @ajschmidt8

### CODE REVIEW

File: ci/test_cpp.sh [hunk 1]
```diff
@@ -35,6 +34,8 @@ rapids-mamba-retry install \
 rapids-logger "Check GPU usage"
 nvidia-smi
 
+EXITCODE=0
+trap "EXITCODE=1" ERR
```
Comment (ajschmidt8) [inline ci/test_cpp.sh:38]: Can you remove the `TODO` a few lines down?

**Review by ajschmidt8 — CHANGES_REQUESTED** [review]

File: ci/test_cpp.sh (hunk 1 above)
Comment (AjayThorve) [reply ci/test_cpp.sh:38]: resolved, thanks!

**Review by ajschmidt8 — APPROVED** [review]
:fire: :fire:

**Comment by ajschmidt8** [comment]
:fire:

**Pull request merged**